tests/test_deidentifier.py

```diff
@@ -543,3 +543,139 @@ def test_missing_dummy_value() -> None:  # noqa
+                }
+            },
+        }
+    )
+    deidentifier.deidentify_dataset(ds)
+
+    sequence = getattr(ds, "ReferencedStudySequence", [])
+    assert len(sequence) == expected_number
+
+
+@pytest.mark.parametrize(
+    "action, expected_value",
+    (
+        (
+            ActionKind.KEEP,
+            "1.2.3",
+        ),
+        (
+            ActionKind.REPLACE,
+            "1.2.3.4.5.6.7.8.9.0.1.2.3.4.5.6.7.8.9.0",
+        ),
+        (
+            ActionKind.REMOVE,
+            None,
+        ),
+    ),
+)
+def test_within_sequence_tag_handling(  # noqa
+    action: ActionKind, expected_value: Optional[str]
+) -> None:
+    ds = Dataset()
+    ds.SOPClassUID = TEST_SOP_CLASS
+
+    # Create a sequence with two items
+    ds.ReferencedStudySequence = pydicom.sequence.Sequence([Dataset()])
+    ds.ReferencedStudySequence[0].ReferencedSOPInstanceUID = "1.2.3"
+
+    deidentifier = DicomDeidentifier(
+        procedure={
+            "sopClass": {
+                TEST_SOP_CLASS: {
+                    "tags": {
+                        tag("ReferencedStudySequence"): {
+                            "default": ActionKind.KEEP
+                        },
+                        tag("ReferencedSOPInstanceUID"): {
+                            "default": action,
+                        },
+                    },
+                }
+            },
+        }
+    )
+    deidentifier.deidentify_dataset(ds)
+
+    if expected_value is None:
+        assert not hasattr(
+            ds.ReferencedStudySequence[0], "ReferencedSOPInstanceUID"
+        )
+    else:
+        value = ds.ReferencedStudySequence[0].ReferencedSOPInstanceUID
+        assert value == expected_value
+
+
+def test_within_sequence_tag_handling_after_replace() -> None:  # noqa
+    ds = Dataset()
+    ds.SOPClassUID = TEST_SOP_CLASS
+
+    # Create a sequence with two items
+    ds.ReferencedStudySequence = pydicom.sequence.Sequence([Dataset()])
+    ds.ReferencedStudySequence[0].ReferencedSOPInstanceUID = "1.2.3"
+    ds.ReferencedStudySequence[0].ReferencedSOPClassUID = "3.4.5"
+
+    deidentifier = DicomDeidentifier(
+        procedure={
+            "sopClass": {
+                TEST_SOP_CLASS: {
+                    "tags": {
+                        tag("ReferencedStudySequence"): {
+                            "default": ActionKind.REPLACE
+                        },
+                        tag("ReferencedSOPInstanceUID"): {
+                            "default": ActionKind.REMOVE,
+                        },
+                    },
+                    "default": ActionKind.KEEP,
+                }
+            },
+        }
+    )
+    deidentifier.deidentify_dataset(ds)
+
+    # The sequence is REPLACEd but the inner tag is REMOVED
+    assert hasattr(ds, "ReferencedStudySequence")
+    value = ds.ReferencedStudySequence[0].ReferencedSOPClassUID
+    assert value == "1.2.3.4.5.6.7.8.9.0.1.2.3.4.5.6.7.8.9.0"  # Dummy value
+    assert not hasattr(
```

Comment:
Why is the inner tag removed? Shouldn't it be replaced as the full sequence is replaced, ignoring the nested actions?

Reply:
The nested actions still apply to the sequence. If a certain tag is not to be there, it should be removed (independent what the outer sequence action describes).